Firebase/InAppMessaging/Flows/FIRIAMDisplayExecutor.m

```diff
@@ -373,6 +373,41 @@ - (void)checkAndDisplayNextContextualMessageForAnalyticsEvent:(NSString *)eventN
   }
 }
 
+- (void)checkAndDisplayNextContextualMessageFromProgrammaticTrigger:(NSString *)triggerName {
+  // synchronizing on self so that we won't potentially enter the render flow from two
+  // threads: example like showing analytics triggered message and a regular app open
+  // triggered message
+  @synchronized(self) {
+    if (self.suppressMessageDisplay) {
+      FIRLogDebug(kFIRLoggerInAppMessaging, @"I-IAM400040",
+                  @"Message display is being suppressed. No message rendering from "
+                  "programmatic trigger.");
+      return;
+    }
+    
+    if (!self.messageDisplayComponent) {
+      FIRLogDebug(kFIRLoggerInAppMessaging, @"I-IAM400041",
+                  @"Message display component is not present yet. No display should happen.");
+      return;
+    }
+    
+    if (self.isMsgBeingDisplayed) {
+      FIRLogDebug(kFIRLoggerInAppMessaging, @"I-IAM400042",
+                  @"An in-app message display is in progress, do not render programmatically "
+                  "triggered message.");
```

Comment:
the triggering doesn't mean there's necessarily a message to render.
perhaps 'in-app message is already displayed. Will not attempt to programmatically trigger '$eventName''. wdyt?